sdk/basyx/aas/adapter/aasx.py

```diff
@@ -203,9 +209,12 @@ def _read_aas_part_into(self, part_name: str,
                     logger.info("Overriding existing object in  ObjectStore with {} ...".format(obj))
                     object_store.discard(obj)
                 else:
-                    logger.warning("Skipping {}, since an object with the same id is already contained in the "
-                                   "ObjectStore".format(obj))
-                    continue
+                    if self.failsafe:
+                        logger.warning("Skipping {}, since an object with the same id is already contained in the "
+                                       "ObjectStore".format(obj))
```

Comment:
```suggestion
                        logger.warning(f"Skipping {obj}, since an object with the same id is already contained in the "
                                       "ObjectStore")
```

I'd stick with one way of formatting strings. I prefer the "f-string" one over the `%s` syntax or the `"".format()`.